### PR TITLE
(more) Shadow integration

### PR DESCRIPTION
**Pull request by jklynch**
This PR adds `beam_statistics_report` to `SirepoDetector`.

### CODE REVIEW

**Comment by mrakitin** [comment]
@jklynch, could you please post a code snippet to use this new feature? Thanks!

**Review by mrakitin — APPROVED** [review]
Looks good! I have a couple of minor comments.

**Comment by jklynch** [comment]
Here is a working example.
```
(sirepo_bluesky) vagrant@vagrant:/vagrant/sirepo-bluesky$ ipython
Python 3.7.10 (default, Jun  4 2021, 14:48:32) 
Type 'copyright', 'credits' or 'license' for more information
IPython 7.24.1 -- An enhanced Interactive Python. Type '?' for help.

In [1]: %run -i examples/prepare_det_env.py
In [2]:    ...: import sirepo_bluesky.sirepo_detector as sd
In [3]:    ...: import bluesky.plans as bp
In [4]:    ...: sirepo_det = sd.SirepoDetector(sim_type='shadow', sim_report_type='shadow_beam_stats', sim_id='9kfDZkjN', sirepo_server='http://localhost:8000')
In [5]:    ...: sirepo_det.select_optic('Aperture')
In [6]:    ...: param1 = sirepo_det.create_parameter('horizontalSize')
In [7]:    ...: param2 = sirepo_det.create_parameter('verticalSize')
In [8]:    ...: sirepo_det.configuration_attrs = ['horizontal_extent', 'vertical_extent', 'shape']
In [9]: RE(bp.count([sirepo_det]))

Transient Scan ID: 1     Time: 2021-06-23 18:00:37
Persistent Unique Scan ID: '79c8d777-27cd-4ec1-bf9e-7adce916f75e'
New stream: 'primary'
+-----------+------------+
|   seq_num |       time |
+-----------+------------+
|         1 | 18:00:42.3 |
+-----------+------------+
generator count ['79c8d777'] (scan num: 1)

Out[9]: ('79c8d777-27cd-4ec1-bf9e-7adce916f75e',)

In [10]: hdr = db[-1]
In [11]: hdr.table().T
Out[11]: 
seq_num                                                                            1
time                                                   2021-06-23 18:00:42.334226608
sirepo_det_image                              1de2438d-da26-4741-8087-7dd9137e5439/0
sirepo_det_shape                                                                 0.0
sirepo_det_mean                                                                  0.0
sirepo_det_photon_energy                                                         0.0
sirepo_det_horizontal_extent                                                     0.0
sirepo_det_vertical_extent                                                       0.0
sirepo_det_sirepo_json             {"models": {"beamStatisticsReport": {"driftDiv...
sirepo_det_beam_statistics_report  [{"isRotated": false, "matrix": [[1.0, 0.0, 0....


```

**Comment by jklynch** [comment]
@mrakitin I added an example (above). I could work on separating the srw and shadow code into separate classes in this PR, or would you rather that had a separate PR?

**Comment by mrakitin** [comment]
Thanks, @jklynch! Let's have that separation in a separate PR.

**Review by AbbyGi — APPROVED** [review]
Looks good.

File: sirepo_bluesky/sirepo_detector.py
Comment (mrakitin) [reply sirepo_bluesky/sirepo_detector.py:140]: Can we remove this piece before merging, please?

**Comment by jklynch** [comment]
@AbbyGi thanks for your review! Would you minding merging?

File: sirepo_bluesky/sirepo_detector.py
Comment (AbbyGi) [reply sirepo_bluesky/sirepo_detector.py:140]: @jklynch could you remove the commented out `hints` first please? I'll merge this after.

**Comment by jklynch** [comment]
@AbbyGi sorry I forgot to remove the hints code

**Comment by AbbyGi** [comment]
Thanks!

File: sirepo_bluesky/sirepo_detector.py
Comment (mrakitin) [reply sirepo_bluesky/sirepo_detector.py]: Thanks!

**Pull request merged**
